File: api-xmlsignature/src/tr/gov/tubitak/uekae/esya/api/xmlsignature/config/Parameters.java
package tr.gov.tubitak.uekae.esya.api.xmlsignature.config;

import org.w3c.dom.Element;

/**
 * @author ayetgin
 */
public class Parameters extends BaseConfigElement
{
    private boolean mWriteReferencedValidationDataToFileOnUpgrade;
    private boolean mAddTimestampValidationData;

    public Parameters()
    {
    }

    public Parameters(Element aElement)
    {
        super(aElement);
        mWriteReferencedValidationDataToFileOnUpgrade = getParamBoolean(ConfigConstants.TAG_WRITE_REFERENCED_PKI_DATA);
        mAddTimestampValidationData = getParamBoolean(ConfigConstants.TAG_ADD_TIMESTAMP_PKI_DATA);
    }

    public boolean isWriteReferencedValidationDataToFileOnUpgrade()
    {
        return mWriteReferencedValidationDataToFileOnUpgrade;
    }
    
    public boolean isAddTimestampValidationData(){
    	return mAddTimestampValidationData;
    }
    
    public void setAddTimestampValidationData(boolean aAddTimestampValidationData){
    	mAddTimestampValidationData = aAddTimestampValidationData;
    }

    public void setWriteReferencedValidationDataToFileOnUpgrade(boolean aWriteReferencedValidationDataToFileOnUpgrade)
    {
        mWriteReferencedValidationDataToFileOnUpgrade = aWriteReferencedValidationDataToFileOnUpgrade;
    }
}

File: api-xmlsignature/csharp/src/config/Parameters.cs
using System.Xml;

namespace tr.gov.tubitak.uekae.esya.api.xmlsignature.config
{
    using Element = XmlElement;

	/// <summary>
	/// @author ayetgin
	/// </summary>
	public class Parameters : BaseConfigElement
	{
		private bool mWriteReferencedValidationDataToFileOnUpgrade;
		private bool mAddTimestampValidationData;

        public Parameters()
        {
            
        }

		public Parameters(Element aElement) : base(aElement)
		{
			mWriteReferencedValidationDataToFileOnUpgrade = getParamBoolean(ConfigConstants.TAG_WRITE_REFERENCED_PKI_DATA);
			mAddTimestampValidationData = getParamBoolean(ConfigConstants.TAG_ADD_TIMESTAMP_PKI_DATA);
		}

		public virtual bool WriteReferencedValidationDataToFileOnUpgrade
		{
			get
			{
				return mWriteReferencedValidationDataToFileOnUpgrade;
			}
            set { mWriteReferencedValidationDataToFileOnUpgrade = value; }
		}

		public virtual bool AddTimestampValidationData
		{
			get
			{
				return mAddTimestampValidationData;
			}
			set
			{
				mAddTimestampValidationData = value;
			}
		}


	}

}